tests/utils.go

```diff
@@ -5238,6 +5239,10 @@ func GetPolicyMatchedToVmi(name string, vmi *v1.VirtualMachineInstance, namespac
 	Expect(name).ToNot(BeEmpty())
 
 	policy := kubecli.NewMinimalMigrationPolicy(name)
+	if policy.Labels == nil {
```

Comment:
I agree that the encapsulation of the cleanup logic is not good. I talked with @fossedihelm about this already, and I think we want something like a TestClient which wraps non-namespaced create calls and manages such labels and handles cleanup. Until then I would want to do it here and not in the `NewMinimalmigrationPolicy` function, because this function has nothing to do with the e2e tests. It just seems to be used to avoid code duplication (which may not be optimal in this case).

Reply:
The client approach sounds good! Thanks for the clarification.